Add Conversion to FiniteDuration to zio.duration

asr2003 opened the pull request:
fixes #8797 
/claim #8797

Hi ZIO Team,

I have added specific exclusions to the MiMa settings in our build.sbt to address the binary compatibility issues reported for the `zio.DurationOps.asScala$extension` and `zio.DurationOps.asScala` methods.

Exclusions Added:
```
ProblemFilters.exclude[IncompatibleResultTypeProblem]("zio.DurationOps.asScala$extension"),
ProblemFilters.exclude[IncompatibleResultTypeProblem]("zio.DurationOps.asScala")
```
The rationale to add this issues to exclude in mima checks is that they are unlikely to impact us because the change would only affect us if someone created a subtype of the extension methods class, which is highly improbable

asr2003 committed:
add FiniteDuration method

asr2003 committed:
modify tests

asr2003 committed:
exclude Zio.Duration in mima checks

asr2003 committed:
minor changes to fix tests

asr2003 committed:
Merge pull request #1 from asr2003/series/2.x

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio?pullRequest=8962) <br/>All committers have signed the CLA.

jdegoes commented:
Closed because this doesn't fix the issue, just as Mima exclusions for a hypothetical violation.

jdegoes committed:
Merge branch 'series/2.x' into add-finite-duration-conversion

asr2003 commented:
@jdegoes Can #8960 be reverted as per https://github.com/zio/zio/pull/8946#issuecomment-2184988465 discussion?

asr2003 committed:
update tests

asr2003 commented:
@jdegoes @kyri-petrou Can this PR be merged quickly to fix the issues of mima checks CI failure in PRS? #8960 doesn't done the work as in the scope of  issue #8797. I have done here and also modified tests to reflect the changes and fixed the mima checks failures

asr2003 committed:
fix lint
